### PR TITLE
feat: add metadata footer to bot messages

### PR DESCRIPTION
**Pull request by rayhanadev**
## Summary
- Adds a `-# ` Discord subtext footer to the end of each bot response showing metadata (e.g. `-# 3.2s · 1,423 tokens · 4 tool calls · 3 steps`)
- Always shows elapsed time; conditionally shows token count (if available), tool calls (if > 0), and steps (if > 1)
- Includes subagent token usage and tool calls, not just the orchestrator's own stats
- Metadata collection is wrapped in try/catch — falls back to elapsed-time-only footer on failure
- Truncation accounts for footer length to stay within Discord's 2000-char limit

## Test plan
- [x] All 23 unit tests pass (`vitest run src/lib/ai/streaming.test.ts`)
- [x] `formatFooter` tested for all field combinations (singular/plural, omission rules, undefined tokens)
- [x] `truncateWithFooter` tested for boundary cases (short text, at-limit, over-limit)
- [ ] Deploy and verify footer appears on bot responses in Discord

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/ai/streaming.ts [hunk 1]
```diff
@@ -125,16 +149,26 @@ export async function streamTurn(
...
   }
 
+  const elapsedMs = Date.now() - startTime;
+  const [totalUsage, steps] = await Promise.all([result.totalUsage, result.steps]);
+  const orchestratorToolCalls = steps.reduce((sum, step) => sum + step.toolCalls.length, 0);
+  const footer = formatFooter({
+    elapsedMs,
+    totalTokens: (totalUsage.totalTokens ?? 0) + subagentMetrics.totalTokens,
+    toolCallCount: orchestratorToolCalls + subagentMetrics.toolCallCount,
+    stepCount: steps.length,
```
[the remark this answers is not in the document]
Comment (rayhanadev) [reply src/lib/ai/streaming.ts]: Fixed in 5803c14 — metadata collection is now wrapped in try/catch, falling back to an elapsed-time-only footer on failure.

File: src/lib/ai/streaming.ts [hunk 2]
```diff
@@ -3,14 +3,36 @@ import type { API } from "@discordjs/core/http-only";
...
+  toolCallCount: number;
+  stepCount: number;
+}): string {
+  const parts: string[] = [];
+  parts.push(`${(meta.elapsedMs / 1000).toFixed(1)}s`);
+  if (meta.totalTokens != null) parts.push(`${meta.totalTokens.toLocaleString("en-US")} tokens`);
+  if (meta.toolCallCount > 0)
+    parts.push(`${meta.toolCallCount} tool call${meta.toolCallCount !== 1 ? "s" : ""}`);
+  if (meta.stepCount > 1) parts.push(`${meta.stepCount} steps`);
+  return `-# ${parts.join(" · ")}`;
```
[the remark this answers is not in the document]
Comment (rayhanadev) [reply src/lib/ai/streaming.ts]: Updated PR description to reflect the omission rules. The conditional display is intentional — showing "0 tool calls" or "1 step" would be noise.

File: src/lib/ai/streaming.ts [hunk 3]
```diff
@@ -3,14 +3,36 @@ import type { API } from "@discordjs/core/http-only";
...
+
+function truncateWithFooter(text: string, footer: string): string {
+  const separator = "\n\n";
+  const available = MAX_LENGTH - footer.length - separator.length;
+  const body = text.length > available ? text.slice(0, available) + "…" : text;
+  return body + separator + footer;
+}
+
 export function truncate(text: string): string {
   return text.length > MAX_LENGTH ? text.slice(0, MAX_LENGTH) + "…" : text;
```
[the remark this answers is not in the document]
Comment (rayhanadev) [reply src/lib/ai/streaming.ts:40]: Added truncateWithFooter tests in 5803c14 covering short text, at-limit, and over-limit cases. Also fixed an off-by-one where the ellipsis wasn't accounted for in the length budget.
